### PR TITLE
Create circular_queue.py

### PR DESCRIPTION
**Pull request by DevilLord9967**
Adding the implementation for Circular queues.

### CODE REVIEW

**Comment by cclauss** [comment]
Please see the doctests at https://github.com/TheAlgorithms/Python/blob/master/maths/greatest_common_divisor.py#L11-L12 and add something similar on each function.

File: data_structures/queue/circular_queue.py [hunk 1]
```diff
@@ -17,6 +23,14 @@ def enqueue(self,data):
 
     #Removing elements from the queue
     def dequeue(self):
+        """
+        Delete the element from the Queue
+        >>> CQ=CircularQueue()
```
Comment (cclauss) [inline data_structures/queue/circular_queue.py]:
```
>>> CQ.size()
0
```

File: data_structures/queue/circular_queue.py [hunk 2]
```diff
@@ -17,6 +23,14 @@ def enqueue(self,data):
...
+        >>> CQ.enqueue(5)
+        True
```
Comment (cclauss) [inline data_structures/queue/circular_queue.py]:
```
>>> CQ.size()
1
```

Comment (DevilLord9967) [reply data_structures/queue/circular_queue.py]: okay

File: data_structures/queue/circular_queue.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        """
+        if self.size()==0:
+            return ("Queue Empty!")
+        data = self.queue[self.head]
+        self.head = (self.head + 1) % self.maxSize
+        return data
+
+    #Calculating the size of the queue
+    def size(self):
+        if self.tail>=self.head:
```
Comment (cclauss) [inline data_structures/queue/circular_queue.py]:
```
        >>> CQ=CircularQueue()
        >>> for i in range(20):
        ...     CQ.enqueue(i)
        ...
        >>> CQ.size() == CQ.max_size

File: data_structures/queue/circular_queue.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
+class CircularQueue:
+
+    #Constructor
+    def __init__(self):
+        self.queue = list()
+        self.head = 0
+        self.tail = 0
+        self.maxSize = 8
```
Comment (cclauss) [inline data_structures/queue/circular_queue.py]: self.max_size = 8

File: data_structures/queue/circular_queue.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        return data
+
+    #Calculating the size of the queue
+    def size(self):
+        if self.tail>=self.head:
+            return (self.tail-self.head)
+        return (self.maxSize - (self.head-self.tail))
+
+q = CircularQueue()
+print(q.enqueue(1))
```
Comment (cclauss) [inline data_structures/queue/circular_queue.py]: Let's do this with a `for` loop.

File: data_structures/queue/circular_queue.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+print(q.enqueue(2))
+print(q.enqueue(3))
+print(q.enqueue(4))
+print(q.enqueue(5))
+print(q.enqueue(6))
+print(q.enqueue(7))
+print(q.enqueue(8))
+print(q.enqueue(9))
+print(q.dequeue())
```
Comment (cclauss) [inline data_structures/queue/circular_queue.py]: Let's do this with a `while q.size():`

**Comment by cclauss** [comment]
Three tests are missing:
1. Immediately after `CQ=CircularQueue()`, `CQ.size() == 0`
2. Even when a lot of items are added to CQ, `CQ.size() <= CQ.max_size`
3. What happens when immediately after `CQ=CircularQueue()`, `CQ.dequeue(self): ` —> error??

**Pull request closed without merging**